mllib/src/main/scala/org/apache/spark/ml/classification/GBTClassifier.scala

```diff
@@ -159,14 +157,21 @@ class GBTClassifier @Since("1.4.0") (
     val numFeatures = oldDataset.first().features.size
     val boostingStrategy = super.getOldBoostingStrategy(categoricalFeatures, OldAlgo.Classification)
 
-    val instr = Instrumentation.create(this, oldDataset)
+    val numClasses: Int = getNumClasses(dataset)
+    if (isDefined(thresholds)) {
```

Comment:
sure, I've added more tests in the latest commit.  I've also fixed an issue where predict was not using thresholds - if they are defined we now use them.